src/main/java/org/opensearch/ad/rest/RestValidateAnomalyDetectorAction.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.ad.rest;
+
+import static org.opensearch.ad.util.RestHandlerUtils.TYPE;
+import static org.opensearch.ad.util.RestHandlerUtils.VALIDATE;
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+
+import org.opensearch.ad.AnomalyDetectorPlugin;
+import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.settings.EnabledSetting;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorAction;
+import org.opensearch.ad.transport.ValidateAnomalyDetectorRequest;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import com.google.common.collect.ImmutableList;
+
+public class RestValidateAnomalyDetectorAction extends AbstractAnomalyDetectorAction {
+    private static final String VALIDATE_ANOMALY_DETECTOR_ACTION = "validate_anomaly_detector_action";
+
+    public RestValidateAnomalyDetectorAction(Settings settings, ClusterService clusterService) {
+        super(settings, clusterService);
+    }
+
+    @Override
+    public String getName() {
+        return VALIDATE_ANOMALY_DETECTOR_ACTION;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of();
+    }
+
+    @Override
+    public List<ReplacedRoute> replacedRoutes() {
+        return ImmutableList
+            .of(
+                // validate detector
+                new ReplacedRoute(
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/%s", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, VALIDATE),
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/%s", AnomalyDetectorPlugin.LEGACY_OPENDISTRO_AD_BASE_URI, VALIDATE)
+                ),
+                // validate detector with type
+                new ReplacedRoute(
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/%s/{%s}", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, VALIDATE, TYPE),
```

Comment:
Can you give an example of an api call hitting this path? And can you add the list of possible `TYPE`s to the PR description? That may be helpful here

Reply:
POST _opendistro/_anomaly_detection/detectors/_validate/model 
or 
POST _opendistro/_anomaly_detection/detectors/_validate/detector will hit this API. 

The payload would be the same but if the type param is model then it will also execute the model "non-blocker" validation on top of the "blocker" detector validation. I didn't add a specific example of that to this PR because this PR doesn't have the implementation for additional model checks but will add something regarding this. Currently if the model type is used then it just runs the detector validation and no additional validation checks for the model at the moment. 

An example response that isn't implemented yet for the model type: 
```
{
  "model": {
    "detection_interval": {
      "suggested_value": {
        "period": {
          "interval": 120,
          "unit": "Minutes"
        }
      }
      "message": "use suggested value"
    }
  }
}
```

Reply:
Got it, have 2 questions:
1. Why does `model` run both detector/blocker validation and model/non-blocker validation? Shouldn't it just run the latter? If users wanted to run both aspects, they could just pass both types (e.g., `<path>/_validate/detector,model`)?
2. Because there is nothing for `model` yet, should that endpoint stay hidden until there's actually additional validation being done? Considering this may go into a separate release before non-blocker validation is done, we don't want to add user confusion there

Reply:
1. Model runs both the detector/blocker and model/non-blocker validation because in order to make sure you can even run the model validation all the detector validation checks have to pass such as there being an existent data index, non negative interval and so on. 

2. I agree that it could be confusing at the moment to have the model param. At the moment I can just keep the detector param only and completely remove the model route.

Reply:
Sounds good, keeping detector param for now and model will just be added in the future